### PR TITLE
fix failing cmaps

### LINKED ISSUE #120: ValueError in terracotta server when connecting with terracotta connect
The error seems to happen, weirdly enough, only when the frontend tries to fetch the magma colormap. All the other colormaps work fine.

Full server log ouput with traceback:

```
(terracotta) λ terracotta serve --debug -d tst.sqlite
 * Serving Flask app "terracotta.server" (lazy loading)
 * Environment: production
   WARNING: Do not use the development server in a production environment.                                                                                   
   Use a production WSGI server instead.                                                                                                                     
 * Debug mode: on                                                                                                                                            
 * Restarting with stat                                                                                                                                      
 * Debugger is active!                                                                                                                                       
 * Debugger PIN: 134-049-503                                                                                                                                 
 * Running on http://localhost:5000/ (Press CTRL+C to quit)                                                                                                  
127.0.0.1 - - [11/Jan/2019 09:41:55] "GET /keys HTTP/1.1" 200 -                                                                                              
127.0.0.1 - - [11/Jan/2019 09:41:56] "GET /swagger.json HTTP/1.1" 200 -                                                                                      
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=greys_r&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                      
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=viridis&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                      
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=rdbu_r&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                       
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=bugn&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                         
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=ylgn&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                         
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=magma&stretch_range=[0,1]&num_values=100 HTTP/1.1" 500 -                                        
Traceback (most recent call last):                                                                                                                           
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 2309, in __call__                             
    return self.wsgi_app(environ, start_response)                                                                                                            
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 2295, in wsgi_app                             
    response = self.handle_exception(e)                                                                                                                      
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 1741, in handle_exception                     
    reraise(exc_type, exc_value, tb)                                                                                                                         
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\_compat.py", line 35, in reraise                            
    raise value                                                                                                                                              
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 2292, in wsgi_app                             
    response = self.full_dispatch_request()                                                                                                                  
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 1815, in full_dispatch_request                
    rv = self.handle_user_exception(e)                                                                                                                       
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 1718, in handle_user_exception                
    reraise(exc_type, exc_value, tb)                                                                                                                         
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\_compat.py", line 35, in reraise                            
    raise value                                                                                                                                              
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 1813, in full_dispatch_request                
    rv = self.dispatch_request()                                                                                                                             
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\flask\app.py", line 1799, in dispatch_request                     
    return self.view_functions[rule.endpoint](**req.view_args)                                                                                               
  File "c:\users\phgr\documents\terracotta\terracotta\server\flask_api.py", line 50, in inner                                                                
    return fun(*args, **kwargs)                                                                                                                              
  File "c:\users\phgr\documents\terracotta\terracotta\server\colormap.py", line 79, in get_colormap                                                          
    payload = {'colormap': colormap(**options)}                                                                                                              
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\contextlib.py", line 52, in inner                                               
    return func(*args, **kwds)                                                                                                                               
  File "c:\users\phgr\documents\terracotta\terracotta\handlers\colormap.py", line 26, in colormap                                                            
    cmap = get_cmap(colormap)                                                                                                                                
  File "c:\users\phgr\documents\terracotta\terracotta\cmaps\__init__.py", line 39, in get_cmap                                                               
    cmap_data = np.load(f)                                                                                                                                   
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\numpy\lib\npyio.py", line 433, in load                            
    pickle_kwargs=pickle_kwargs)                                                                                                                             
  File "C:\Users\phgr\AppData\Local\Continuum\Miniconda2\envs\terracotta\lib\site-packages\numpy\lib\format.py", line 698, in read_array                     
    array.shape = shape                                                                                                                                      
ValueError: cannot reshape array of size 764 into shape (255,3)                                                                                              
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=gist_earth&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -                                   
127.0.0.1 - - [11/Jan/2019 09:41:59] "GET /colormap?colormap=ocean&stretch_range=[0,1]&num_values=100 HTTP/1.1" 200 -
```

### PR DESCRIPTION
**Pull request by j08lue**
Closes #120 and adds test to prevents this from happening again.

### CODE REVIEW

**Comment by j08lue** [comment]
> Double-check that npy files are portable

Right. From https://docs.scipy.org/doc/numpy-1.13.0/reference/generated/numpy.save.html:

> allow_pickle : bool, optional
>    Allow saving object arrays using Python pickles. Reasons for disallowing pickles include security (loading pickled data can execute arbitrary code) and portability (pickled objects may not be loadable on different Python installations, for example if the stored objects require libraries that are not available, and not all pickled data is compatible between Python 2 and Python 3). Default: True

So maybe `allow_pickle=False` does it.

**Comment by j08lue** [comment]
No, whether I set `allow_pickle=False` or not makes no difference to the files.

**Comment by j08lue** [comment]
But the tests fail due to some issue in `pytest_benchmark`. I guess that is on their side and I should raise it?

**Comment by dionhaefner** [comment]
Looks like an incompatibility with pytest-flake8. I'd be grateful of someone looked into it; we could raise it and pin the version until fixed.

**Comment by j08lue** [comment]
No, `pytest-flake8` has not changed in half a year. This is one is on `pytest-benchmark`.

**Comment by dionhaefner** [comment]
Could you try again with forced binary encoding for the npy files? I'd prefer `np.load` and `np.save` if we can make it work.

**Comment by j08lue** [comment]
Binary files seem to work fine and are perhaps just the bit we need more portable than `.npy` and since they all have the same constant shape and data type, we are good I'd say.

**Comment by j08lue** [comment]
> Could you try again with forced binary encoding for the npy files?

OK. Did not see that `np.load` has `allow_pickle`, too. I'll give it one more shot.

File: setup.py [hunk 1]
```diff
@@ -77,7 +77,7 @@
             'pytest-cov',
             'pytest-mypy',
             'pytest-flake8',
-            'pytest-benchmark',
+            'pytest-benchmark==3.1.1',
```
Comment (dionhaefner) [inline setup.py]: I think we mean `'pytest-benchmark!=3.1.2'`

File: terracotta/cmaps/__init__.py [hunk 2]
```diff
@@ -36,7 +36,7 @@ def get_cmap(name: str) -> np.ndarray:
         raise ValueError(f'Unknown colormap {name}, must be one of {AVAILABLE_CMAPS}')
 
     with _get_cmap_data(name) as f:
-        cmap_data = np.load(f)
+        cmap_data = np.fromfile(f, dtype=np.uint8).reshape((255, 3))
```
Comment (dionhaefner) [inline terracotta/cmaps/__init__.py]: See comment above; could we give `.npy` files one more shot with the additions to `.gitattributes`?

File: setup.py (hunk 1 above)
Comment (j08lue) [reply setup.py]: Probably, yes. I'll revisit that soon. #123

**Pull request closed without merging**